OpenSim/Simulation/SimbodyEngine/Joint.h

```diff
@@ -131,30 +113,65 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Joint, ModelComponent);
 
         @param[in] name     the name associated with this joint (should be
                             unique from other joints in the same model)
-        @param[in] parent   the parent PhysicalFrame that joint connects to
-        @param[in] locationInParent    Vec3 of the location of the joint in the
-                                       parent body frame.
-        @param[in] orientationInParent Vec3 of the XYZ body-fixed Euler angles
-                                       of the joint frame orientation in the
-                                       parent body frame.
-        @param[in] child    the child PhysicalFrame that joint connects to
-        @param[in] locationInChild     Vec3 of the location of the joint in the
-                                       child body frame.
-        @param[in] orientationInChild  Vec3 of the XYZ body-fixed Euler angles
-                                       of the joint frame orientation in the
-                                       child body frame.
+        @param[in] parentName   the name of the parent PhysicalFrame of the joint
+        @param[in] childName    the name of the child PhysicalFrame of the joint
         @param[in] reverse  Advanced optional flag (bool) specifying the 
                             direction of the Joint in the multibody tree. 
-                            Default is false (that is, forward).
+                            Default is false (that is, forward. parent to child).
         */
-    Joint(const std::string &name,
-          const PhysicalFrame& parent,
-          const SimTK::Vec3& locationInParent,
-          const SimTK::Vec3& orientationInParent,
-          const PhysicalFrame& child,
-          const SimTK::Vec3& locationInChild,
-          const SimTK::Vec3& orientationInChild,
-          bool reverse = false);
+    Joint( const std::string& name,
+           const std::string& parentName,
+           const std::string& childName,
+           bool reverse = false);
+
+    /** DEPRECATED Convenience Constructor
```

Comment:
We should have a consistent convention; @sherm1 uses `<b>(Deprecated)</b>`: https://github.com/simbody/simbody/blob/master/SimTKcommon/include/SimTKcommon/internal/ClonePtr.h#L293